### PR TITLE
Add lib support to zwave-js 9.6.0 and upstream server changes

### PR DESCRIPTION
**Pull request by raman325**
9.6.0 introduced some new parameters to the Entry Control CC And Multilevel Switch CC notification events: https://github.com/zwave-js/node-zwave-js/releases/tag/v9.6.0

All other changes/additions are captured here: https://github.com/zwave-js/zwave-js-server/pull/685 and https://github.com/zwave-js/zwave-js-server/pull/686

### CODE REVIEW

File: zwave_js_server/model/firmware.py [hunk 1]
```diff
@@ -153,3 +153,65 @@ def status(self) -> FirmwareUpdateStatus:
...
+
+class FirmwareUpdateFileInfoDataType(TypedDict):
+    """Represent a firmware update file info data dict type."""
+
+    target: int
+    url: str
+    integrity: str  # sha256
+
+
+class FirmwareUpdateFileInfo:
```
Comment (MartinHjelmare) [inline zwave_js_server/model/firmware.py:166]: This can be a dataclass.

File: zwave_js_server/model/firmware.py [hunk 2]
```diff
@@ -153,3 +153,65 @@ def status(self) -> FirmwareUpdateStatus:
...
+
+class FirmwareUpdateInfoDataType(TypedDict):
+    """Represent a firmware update info data dict type."""
+
+    version: str
+    changelog: str
+    files: List[FirmwareUpdateFileInfoDataType]
+
+
+class FirmwareUpdateInfo:
```
Comment (MartinHjelmare) [inline zwave_js_server/model/firmware.py:197]: This too.

File: zwave_js_server/model/firmware.py (hunk 1 above)
Comment (raman325) [reply zwave_js_server/model/firmware.py:166]: agreed

File: zwave_js_server/model/firmware.py [hunk 3]
```diff
@@ -194,24 +181,19 @@ class FirmwareUpdateInfoDataType(TypedDict):
...
-        """Return the firmware files."""
-        return [FirmwareUpdateFileInfo(file) for file in self.data["files"]]
+    version: str
+    changelog: str
+    files: List[FirmwareUpdateFileInfo]
+
+    @classmethod
+    def from_dict(cls, data: FirmwareUpdateInfoDataType) -> "FirmwareUpdateInfo":
+        """Initialize from dict."""
+        return FirmwareUpdateInfo(
```
Comment (MartinHjelmare) [inline zwave_js_server/model/firmware.py:195]:
```suggestion
        return cls(
```

**Review by MartinHjelmare — APPROVED** [review]

**Comment by MartinHjelmare** [comment]
Should we merge here?

**Pull request merged**